tests/integration/devfile/utils/utils.go

```diff
@@ -585,3 +586,28 @@ func DeleteLocalConfig(args ...string) {
 	}
 	helper.MatchAllInOutput(output, expectedOutput)
 }
+
+// VerifyCatalogListComponent verifies component exists in both S2I Component list and Devfile Component list
+func VerifyCatalogListComponent(output string, componentName []string) error {
+	var data map[string]interface{}
+	listItems := []string{"devfileItems"}
+
+	err := json.Unmarshal([]byte(output), &data)
+	if err != nil {
+		return err
+	}
+
+	if os.Getenv("KUBERNETES") != "true" {
+		listItems = append(listItems, "s2iItems")
+	}
+
+	for _, items := range listItems {
+		outputBytes, err := json.Marshal(data[items])
+		if err != nil {
+			return err
+		}
+		output = string(outputBytes)
+		helper.MatchAllInOutput(output, componentName)
+	}
+	return err
```

Comment:
```return nil```

Reply:
It will be same 👍

Reply:
Nope it not same. err is returned in the block level itself when it is hit for first time. 

All the time the function will return the error of the statment ```err := json.Unmarshal([]byte(output), &data)```